webapi/src/main/java/org/sakaiproject/webapi/controllers/DashboardController.java

```diff
@@ -157,13 +160,15 @@ public DashboardRestBean getUserDashboard(@PathVariable String userId) throws Us
         try {
             List<AnnouncementMessage> motdMessages = announcementService.getMessages(
                 announcementService.getSummarizableReference(null, announcementService.MOTD_TOOL_ID),
-                new ViewableFilter(null, null, 1, announcementService),
+                new ViewableFilter(null, null, maxNumberMotd, announcementService),
                 false,
                 false);
 
-            if (motdMessages.size() > 0) {
-                bean.setMotd(motdMessages.get(motdMessages.size() - 1).getBody());
+            StringBuffer sb = new StringBuffer();
+            for (AnnouncementMessage motdMessage : motdMessages) {
+                sb.append(motdMessage.getBody());
```

Comment:
You could get rid of  the motdMessages var here, just do a forEach on the getMessages result to build the string.

Reply:
yes but it's so readddddable now

Reply:
StringBuilder sb = new StringBuilder();
announcementService.getMessages(
                announcementService.getSummarizableReference(null, announcementService.MOTD_TOOL_ID),
                new ViewableFilter(null, null, maxNumberMotd, announcementService),
                false, false).forEach(msg -> sb.append(msg.getBody());
return sb.toString();

That's not too bad! I think using reduce would be going a bit far, but this looks fine to me, and no need for the variable. But, it's up to you. I've done so much js and scala that this style seems clear to me, but maybe that's just me.

Reply:
Looks okay but it doesn't compile ;)

I can't figure it out .....

Reply:
It's because the return from getMessages is just an untyped List. The hierarchy in message service and announcement service is a total maze and doesn't really lend itself to this functional approach, so .. I'd just merge your original. I'd not want to start messing with the class hierarchy in announcement etc. Sorry I pulled you off track :(